src/HttpClientFactory/Http/src/Logging/HttpHeadersLogValue.cs

```diff
@@ -90,10 +92,27 @@ public override string ToString()
 
                 for (var i = 0; i < Values.Count; i++)
                 {
+                    var isLogSensitiveHeader = false;
                     var kvp = Values[i];
                     builder.Append(kvp.Key);
                     builder.Append(": ");
 
+                    for (int headerIndex = 0; headerIndex < _logSensitiveHeaders.Count; headerIndex++)
```

Comment:
I think we'd really like to avoid the O(MN) here nested loop in the case that the list of logged headers is really large. I think this could really be simpler and more performant with a hashset.